src/Lykke.AzureStorage.Test/TableStorageEntity/Metamodel/Providers/CompositeMetamodelProviderTest.cs

```diff
@@ -0,0 +1,288 @@
+﻿using System;
+using System.Reflection;
+using JetBrains.Annotations;
+using Lykke.AzureStorage.Tables.Entity.Metamodel.Providers;
+using Lykke.AzureStorage.Tables.Entity.Serializers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+
+namespace Lykke.AzureStorage.Test.TableStorageEntity.Metamodel.Providers
+{
+    [TestClass]
+    public class CompositeMetamodelProviderTest
+    {
+        #region Mocks
+
+        private class ValueSerializerMock : IStorageValueSerializer
+        {
+            public string Serialize(object value)
+            {
+                throw new NotImplementedException();
+            }
+
+            public object Deserialize(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class AnotherValueSerializerMock : IStorageValueSerializer
+        {
+            public string Serialize(object value)
+            {
+                throw new NotImplementedException();
+            }
+
+            public object Deserialize(string serialized)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        private class TestType
+        {
+            [UsedImplicitly]
+            public int Property { get; set; }
+        }
+
+        #endregion
+
+
+        #region Type
+
+        [TestMethod]
+        public void Test_that_null_is_returned_if_all_providers_return_no_serializer_when_getting_type_serializer()
+        {
+            var nestedProvider1Mock = new Mock<IMetamodelProvider>();
+            var nestedProvider2Mock = new Mock<IMetamodelProvider>();
+
+            nestedProvider1Mock
+                .Setup(x => x.TryGetTypeSerializer(It.IsAny<Type>()))
+                .Returns<Type>(t => null);
+
+            nestedProvider2Mock
+                .Setup(x => x.TryGetTypeSerializer(It.IsAny<Type>()))
+                .Returns<Type>(t => null);
+
+            var provider = new CompositeMetamodelProvider();
+            IMetamodelProvider metamodelProvider = provider;
+            provider
+                .AddProvider(nestedProvider1Mock.Object)
+                .AddProvider(nestedProvider2Mock.Object);
+
+            // Act
+            var serializer = metamodelProvider.TryGetTypeSerializer(typeof(TestType));
+
+            // Assert
+            Assert.IsNull(serializer);
+        }
+
+        [TestMethod]
+        public void Test_that_provider_used_in_the_order_wich_they_was_registered_when_getting_type_serializer()
```

Comment:
Typo: wich

Reply:
Will fix in the next PR to prevent merging conflicts